Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForestState.java

```diff
@@ -15,55 +15,54 @@
 
 package com.amazon.randomcutforest;
 
-import java.util.ArrayList;
 import java.util.List;
 
+import lombok.Data;
+
 import com.amazon.randomcutforest.executor.Sequential;
-import com.amazon.randomcutforest.sampler.CompactSamplerData;
 import com.amazon.randomcutforest.sampler.Weighted;
-import com.amazon.randomcutforest.store.PointStoreDoubleData;
-import com.amazon.randomcutforest.tree.TreeData;
+import com.amazon.randomcutforest.state.sampler.CompactSamplerState;
+import com.amazon.randomcutforest.state.store.PointStoreDoubleState;
+import com.amazon.randomcutforest.tree.CompactRandomCutTreeState;
 
 /**
  * A class that encapsulates most of the data used in a RandomCutForest such
  * that the forest can be serialized and deserialized.
  */
-public class ForestState {
-
-    public String typeOfForest = null;
+@Data
+public class RandomCutForestState {
 
-    public long entreesSeen;
+    private long entreesSeen;
 
-    public double lambda;
+    private double lambda;
 
-    public int numberOfTrees;
+    private int numberOfTrees;
 
-    public int sampleSize;
+    private int sampleSize;
 
-    public int dimensions;
+    private int dimensions;
 
-    public int outputAfter;
+    private int outputAfter;
 
-    public boolean storeSequenceIndexesEnabled;
+    private boolean storeSequenceIndexesEnabled;
 
-    public boolean compactEnabled;
+    private boolean compactEnabled;
 
-    public boolean centerOfMassEnabled;
+    private boolean centerOfMassEnabled;
 
-    public boolean parallelExecutionEnabled;
+    private boolean parallelExecutionEnabled;
 
-    public int threadPoolSize;
+    private int threadPoolSize;
 
     public boolean saveTreeData;
 
-    public PointStoreDoubleData pointStoreDoubleData;
-
-    public ArrayList<CompactSamplerData> compactSamplerData;
+    public PointStoreDoubleState pointStoreDoubleState;
 
-    public ArrayList<List<Sequential<double[]>>> sequentialSamplerData;
+    public List<CompactSamplerState> compactSamplerStates;
 
-    public ArrayList<List<Weighted<double[]>>> smallSamplerData;
+    public List<List<Sequential<double[]>>> sequentialSamplerData;
 
-    public ArrayList<TreeData> treeData;
+    public List<List<Weighted<double[]>>> smallSamplerData;
```

Comment:
question. are these two variables both for current rcf? the difference is whether there is sequence or not?

they might use a better name to be differentiated from other sample/state.

Reply:
With our new plan of converting a list of noncompact samplers into a pointstorestate and a list of compactsamplerstates, these should eventually be removed.